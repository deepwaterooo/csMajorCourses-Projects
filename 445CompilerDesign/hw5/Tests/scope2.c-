// C-F13
// C-F09
int g;
int ant(int x)
{
    int a;
    int b;
    b=4;
    x=5;
    g=6;
    {
	int c;
        int b;
	a=6;
	b=7;
	c=8;
	g=9;
	x=10;
        output(a);
        output(b);
        output(c);
        output(g);
        output(x);
        outnl();
    }
    output(a);
    output(b);
    output(g);
    output(x);
    outnl();
    a=11;
    b=12;
    x=14;
    {
	x=15;
    }
    output(a);
    output(b);
    output(g);
    output(x);
    outnl();

    return a+b+g+x;
}

  main()
{
    int a;
    a=16;
    g=1;
    output(ant(2));
    output(g);
    output(a);
    outnl();
}
