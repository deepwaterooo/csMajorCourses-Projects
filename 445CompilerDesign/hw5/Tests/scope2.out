ERROR(13): Symbol 'b' is already defined at line 7.
Number of warnings: 0
Number of errors: 1
------------------------------------
